src/config.rs

```diff
@@ -224,30 +284,42 @@ pub async fn remappings() -> Result<(), ConfigError> {
         existing_remap.push(String::from(split[0]));
     });
 
-    dependencies.iter().for_each(|dependency| {
-        let mut dependency_name_formatted = format!("{}-{}", &dependency.name, &dependency.version);
-        if !dependency_name_formatted.contains('@') {
-            dependency_name_formatted = format!("@{}", dependency_name_formatted);
-        }
+    if soldeer_config.reg_remappings {
+        let dependencies: Vec<Dependency> = match read_config_deps(config_file).await {
+            Ok(dep) => dep,
+            Err(err) => {
+                return Err(err);
+            }
+        };
+
+        dependencies.iter().for_each(|dependency| {
+            let dependency_name_formatted = format_remap_name(soldeer_config, dependency);
+            let index = existing_remap.iter().position(|r| r == &dependency_name_formatted);
+            if index.is_none() {
+                Paint::green(&format!(
+                    "Added a new dependency to remappings {}-{}",
+                    dependency.name, dependency.version
+                ));
+                new_remappings.push_str(&format!(
+                    "\n{}=dependencies/{}-{}/",
+                    dependency_name_formatted, dependency.name, dependency.version
+                ));
+            }
+        });
```

Comment:
In this case, we should probably ignore all existing remappings and re-generate everything